Add release notes for 2.8.0 (#8472)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## Release 2.8.0

### Work Pools 🎉
- Remove experimental designation around work pools — https://github.com/PrefectHQ/prefect/pull/8362
- Adds work pool and queue filters to flow runs history endpoint — https://github.com/PrefectHQ/prefect/pull/8459
- Update flowRunFilter on WorkPool page — https://github.com/PrefectHQ/prefect/pull/8453

> - Updates agent logic to respect work queue priority — https://github.com/PrefectHQ/prefect/pull/8458

Author: To be updated

### Enhancements

> - Add ability to create a new flow run from the UI with custom default parameter values taken from a previous run — https://github.com/PrefectHQ/prefect/pull/8405

Reviewer: Maybe a more succinct name? Not a strong preference
```suggestion
- Add ability to create a flow run from the UI with parameters from a previous run — https://github.com/PrefectHQ/prefect/pull/8405
```

- Add generic `Webhook` block — https://github.com/PrefectHQ/prefect/pull/8401
- Add override customizations functionality to deployments via CLI — https://github.com/PrefectHQ/prefect/pull/8349
- Add ability to reset concurrency limits in CLI to purge existing runs from taking concurrency slots — https://github.com/PrefectHQ/prefect/pull/8408
- Ensure matching flow run state information in UI — https://github.com/PrefectHQ/prefect/pull/8441
- Customize CLI block registration experience based on `PREFECT_UI_URL` — https://github.com/PrefectHQ/prefect/pull/8438

### Fixes
- Fix `prefect dev start` command — https://github.com/PrefectHQ/prefect/pull/8176

> - Fix ability to see long log messages when in Prefect UI — https://github.com/PrefectHQ/prefect/pull/8449

Reviewer: ```suggestion
- Fix display of long log messages when in the UI — https://github.com/PrefectHQ/prefect/pull/8449
```

- Update `get_run_logger` to accomodate returning `logging.LoggerAdapter` — https://github.com/PrefectHQ/prefect/pull/8422
- Restore Prefect wrapper around HTTP errors for nicer error messages — https://github.com/PrefectHQ/prefect/pull/8391

### Documentation
- Update Infrastructure concept documentation with extra-pip-package example and updated deployment.yaml — https://github.com/PrefectHQ/prefect/pull/8465

> - Ensure Prefect docs UI components compatible with Material for Mkdocs 9.0.10 — https://github.com/PrefectHQ/prefect/pull/8403

Reviewer: Probably a `maintenance` item and could be excluded.

# Prefect Release Notes

## Release 2.8.0

### Prioritize flow runs with work pools 🏊

![Work pools allow you to organize and prioritize work](https://user-images.githubusercontent.com/12350579/217914094-e8064420-294b-4033-b12e-c0f58da521d5.png)

With this release, flow runs can now be prioritized between work queues via work pools! Work pools allow you to organize and prioritize work by grouping related work queues together. Within work pools, you can assign a priority to each queue, and flow runs scheduled on higher priority work queue will be run before flow runs scheduled on lower priority work queues. This allows agents to prioritize work that is more important or time-sensitive even if there is a large backlog of flow runs on other work queues in a given work pool.

> All existing work queues will be assigned to a default work pool named `default-agent-pool`. Creating a new work pool can be done via the Work Pool tab in the UI or the CLI.

Reviewer 2: ```suggestion
All existing work queues will be assigned to a default work pool named `default-agent-pool`. Creating a new work pool can be done via the Work Pools page in the UI or the CLI.
```

To create a new work queue in a work pool via the CLI:

```bash
prefect work-queue create "high-priority" --pool "my-pool"
```

Deployments can now be assigned to a work queue in a specific work pool. Use the `--pool` flag to specify the work pool and the `--queue` flag to specify the work queue when building a deployment.

```bash
prefect deployment build \
    --pool my-pool \
    --queue high-priority \   
    --name high-priority \
    high_priority_flow.py:high_priority_flow
```

Once a deployment has been created and is scheduling flow runs on a work queue, you can start an agent to pick up those flow runs by starting an agent with the `--pool` flag.

```bash
prefect agent start --pool my-pool
```

Starting an agent with the `--pool` command will allow the agent to pick up flow runs for the entire pool even as new queues are added to the pool. If you want to start an agent that only picks up flow runs for a specific queue, you can use the `--queue` flag.

```bash
prefect agent start --pool my-pool --queue high-priority
```

To learn more about work pools, check out the [docs](https://docs.prefect.io/concepts/work-pools/) or see the relevant pull requests:

> - Add ability to filter on work pool and queue when viewing flow run history — https://github.com/PrefectHQ/prefect/pull/8459

Reviewer 2: ```suggestion

- Add ability to filter on work pool and queue when viewing flow run history — https://github.com/PrefectHQ/prefect/pull/8459
```

- Add ability to reset concurrency limits in CLI to purge existing runs from taking concurrency slots — https://github.com/PrefectHQ/prefect/pull/8408
- Ensure matching flow run state information in UI — https://github.com/PrefectHQ/prefect/pull/8441
- Customize CLI block registration experience based on `PREFECT_UI_URL` — https://github.com/PrefectHQ/prefect/pull/8438

### Fixes
- Fix `prefect dev start` command — https://github.com/PrefectHQ/prefect/pull/8176
- Fix display of long log messages when in the UI — https://github.com/PrefectHQ/prefect/pull/8449
- Update `get_run_logger` to accomodate returning `logging.LoggerAdapter` — https://github.com/PrefectHQ/prefect/pull/8422
- Restore Prefect wrapper around HTTP errors for nicer error messages — https://github.com/PrefectHQ/prefect/pull/8391
- Fix display of work pool flow run filter in the UI — https://github.com/PrefectHQ/prefect/pull/8453

### Documentation
- Update Infrastructure concept documentation with `extra-pip-package` example and updated `deployment.yaml` — https://github.com/PrefectHQ/prefect/pull/8465

> 

Reviewer 2: ```suggestion
- Add Work Pools documentation - https://github.com/PrefectHQ/prefect/pull/8377

```

Author: thanks!

```bash
prefect deployment build \
    --pool my-pool \
    --queue high-priority \   
    --name high-priority \
    high_priority_flow.py:high_priority_flow
```

Once a deployment has been created and is scheduling flow runs on a work queue, you can start an agent to pick up those flow runs by starting an agent with the `--pool` flag.

```bash
prefect agent start --pool my-pool
```

Starting an agent with the `--pool` command will allow the agent to pick up flow runs for the entire pool even as new queues are added to the pool. If you want to start an agent that only picks up flow runs for a specific queue, you can use the `--queue` flag.

```bash
prefect agent start --pool my-pool --queue high-priority
```

To learn more about work pools, check out the [docs](https://docs.prefect.io/concepts/work-pools/) or see the relevant pull requests:

- Add ability to filter on work pool and queue when querying flow runs — https://github.com/PrefectHQ/prefect/pull/8459
- Remove experimental designation around work pools — https://github.com/PrefectHQ/prefect/pull/8362

> - Ensure agent respects work queue priority — https://github.com/PrefectHQ/prefect/pull/8458

Reviewer: These can be listed in the normal `enhancements` section.

Author: I merged right after approval but added those changes here:
https://github.com/PrefectHQ/prefect/pull/8474
